### ISSUE TITLE
Delay in receiving location on getCurrentLocation() when precise location not granted [IOS]

### ISSUE THREAD

**Issue by newtracs** [transistorsoft/react-native-background-geolocation#1419]
## Your Environment
* Plugin version: 4.4.3
* Platform: iOS
* OS version: 15.1.1
* Device manufacturer / model: Apple iPhone 12 Mini
* React Native version (`react-native -v`): 0.64
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->

const backgroundConfig = {
  ...(isIOS ? {
    stationaryRadius: 5,
    distanceFilter: 5,
  } : {
    distanceFilter: 0,
    locationUpdateInterval: 2000,
    fastestLocationUpdateInterval: 1000,
  }),
  desiredAccuracy: BackgroundGeolocation.HIGH,
  desiredOdometerAccuracy: 100,
  elasticityMultiplier: 2,
  stopOnStationary: true,
  stopOnTerminate: false,
  foregroundService: true,
  //startOnBoot: true,
  //enableHeadless: true,
  preventSuspend: true,
  //debug: true,
  autoSync: false,
  logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
  locationAuthorizationRequest: 'WhenInUse',
  backgroundPermissionRationale: {
    title: "Allow {applicationName} to access to this device's location in the background?",
    message: "In order to log trips in the background, please enable '{backgroundPermissionOptionLabel}' location permission",
    positiveAction: "Change to '{backgroundPermissionOptionLabel}'",
    negativeAction: "Cancel"
  },
  //disableLocationAuthorizationAlert: true,
  locationAuthorizationAlert: {
    titleWhenNotEnabled: "Location services are not enabled",
    titleWhenOff: "Location services are not enabled",
    instructions: "To ensure best app experience, please enable 'Always' location permissions in device settings",
    cancelButton: "Cancel",
    settingsButton: "Settings"
  }
}
```

## Expected Behavior

When precise location accuracy is denied in iOS 14+, getCurrentPosition() should return a location almost immediately.

## Actual Behavior

When precise location accuracy is denied in iOS 14+, getCurrentPosition() takes much longer to return a result when requested, at least 60 seconds when testing on a real device & in the simulator.

## Steps to Reproduce

1.  Code to request location:

```javascript 
const panToUserLocation = async () => {
    try {
      console.log('obtaining user location')
      let location = await BackgroundGeolocation.getCurrentPosition()
      console.log('obtaining user location...done').  //<-- takes around  1 minute to return location
      const { longitude, latitude } = location.coords
      panTo(longitude, latitude)
    } catch (err) {
      console.log('Pan to user location error', err)
    }
  }
```

## Context

I am in the process of managing app behaviour for different location permission scenarios and how to respond when user denies precise location accuracy.

## Debug logs

<details><summary>Logs</summary>

``` <!-- Syntax highlighting:  DO NOT REMOVE -->
2021-12-08 14:38:31.171415+1100 Newtracs[430:11664] [javascript] '[requestTemporaryFullAccuracy] DENIED: ', 1
2021-12-08 14:38:41.998667+1100 Newtracs[430:11664] [javascript] obtaining user location
2021-12-08 14:39:42.007424+1100 Newtracs[430:11664] [javascript] obtaining user location...done
```

</details>

**Comment by newtracs** [comment]
I figured out the issue here halfway through writing this so will provide my solution for those who come across it.  I needed to tweak the config for the getCurrentPosition() call, which was all default settings until now.

This is in the docs --> https://transistorsoft.github.io/react-native-background-geolocation/interfaces/currentpositionrequest.html#samples

Essentially, the plugin will wait for the amount of time set in `timeout` until it gets a point within the `accuracy` parameter.  Since precise accuracy is not allowed, it was waiting 30s each time then reverting to the last location available.

My new config is:

```javascript
export const getLocationConfig = {
    timeout: 1,          // 30 second timeout to fetch location
    persist: false,        // Defaults to state.enabled
    maximumAge: 5000,     // Accept the last-known-location if not older than 5000 ms.
    desiredAccuracy: 10,  // Try to fetch a location with an accuracy of `10` meters.
    samples: 5,
  }
```
